test-suite/WhichLanguage_test.go

```diff
@@ -24,11 +24,18 @@ func TestWhichLanguage(t *testing.T) {
 		template := bt.Backend.Name + "/one-dep/"
 
 		bt.Subtest(bt.Backend.Name, func(bt testUtils.BackendT) {
-			bt.Subtest("locked", func(bt testUtils.BackendT) {
-				bt.AddTestFile(template+bt.Backend.Specfile, bt.Backend.Specfile)
-				bt.AddTestFile(template+bt.Backend.Lockfile, bt.Backend.Lockfile)
-				bt.UpmWhichLanguage()
-			})
+			if bt.Backend.QuirksIsReproducible() {
+				bt.Subtest("locked", func(bt testUtils.BackendT) {
+					bt.AddTestFile(template+bt.Backend.Specfile, bt.Backend.Specfile)
+					bt.AddTestFile(template+bt.Backend.Lockfile, bt.Backend.Lockfile)
+					bt.UpmWhichLanguage()
+				})
+			} else {
+				bt.Subtest("no lockfile", func(bt testUtils.BackendT) {
+					bt.AddTestFile(template+bt.Backend.Specfile, bt.Backend.Specfile)
+					bt.UpmWhichLanguage()
+				})
+			}
```

Comment:
i think we can just get rid of this `else`, there's nothing different vs the default case. can't exactly test multiple non-reproducible package managers that use the same spec file 😬

Reply:
Sorry, maybe missing something -- the `default` case only curently applies to `bun`, are you suggesting I add `python3-pip` to that list? "default" doesn't really mean anything useful in the pip case, do you have insight into what it could be generalized into?

Reply:
ah, you're right, lgtm